administrator/components/com_finder/src/Indexer/Debugadapter.php

```diff
@@ -0,0 +1,979 @@
+	{
+		$query = $this->db->getQuery(true)
+			->select($this->db->quoteName('access'))
+			->from($this->db->quoteName($this->table))
+			->where($this->db->quoteName('id') . ' = ' . (int) $row->id);
+		$this->db->setQuery($query);
+
+		// Store the access level to determine if it changes
+		$this->old_access = $this->db->loadResult();
+	}
+
+	/**
+	 * Method to get the number of content items available to index.
+	 *
+	 * @return  integer  The number of content items available to index.
+	 *
+	 * @since   __DEPLOY_VERSION__
+	 * @throws  Exception on database error.
+	 */
+	protected function getContentCount()
+	{
+		$return = 0;
+
+		// Get the list query.
+		$query = $this->getListQuery();
+
+		// Check if the query is valid.
+		if (empty($query))
+		{
+			return $return;
+		}
+
+		// Tweak the SQL query to make the total lookup faster.
+		if ($query instanceof QueryInterface)
+		{
+			$query = clone $query;
+			$query->clear('select')
+				->select('COUNT(*)')
+				->clear('order');
+		}
+
+		// Get the total number of content items to index.
+		$this->db->setQuery($query);
+
+		return (int) $this->db->loadResult();
+	}
+
+	/**
+	 * Method to get a content item to index.
+	 *
+	 * @param   integer  $id  The id of the content item.
+	 *
+	 * @return  Result  A Result object.
+	 *
+	 * @since   __DEPLOY_VERSION__
+	 * @throws  Exception on database error.
+	 */
+	protected function getItem($id)
+	{
+		// Get the list query and add the extra WHERE clause.
+		$query = $this->getListQuery();
+		$query->where('a.id = ' . (int) $id);
+
+		// Get the item to index.
+		$this->db->setQuery($query);
+		$item = $this->db->loadAssoc();
+
+		// Convert the item to a result object.
+		$item = ArrayHelper::toObject((array) $item, Result::class);
+
+		// Set the item type.
+		$item->type_id = $this->type_id;
+
+		// Set the item layout.
+		$item->layout = $this->layout;
+
+		return $item;
+	}
+
+	/**
+	 * Method to get a list of content items to index.
+	 *
+	 * @param   integer         $offset  The list offset.
+	 * @param   integer         $limit   The list limit.
+	 * @param   QueryInterface  $query   A QueryInterface object. [optional]
+	 *
+	 * @return  Result[]  An array of Result objects.
+	 *
+	 * @since   __DEPLOY_VERSION__
+	 * @throws  Exception on database error.
+	 */
+	protected function getItems($offset, $limit, $query = null)
+	{
+		// Get the content items to index.
+		$this->db->setQuery($this->getListQuery($query)->setLimit($limit, $offset));
+		$items = $this->db->loadAssocList();
+
+		foreach ($items as &$item)
+		{
+			$item = ArrayHelper::toObject($item, Result::class);
+
+			// Set the item type.
+			$item->type_id = $this->type_id;
+
+			// Set the mime type.
+			$item->mime = $this->mime;
+
+			// Set the item layout.
+			$item->layout = $this->layout;
+		}
+
+		return $items;
+	}
+
+	/**
+	 * Method to get the SQL query used to retrieve the list of content items.
+	 *
+	 * @param   mixed  $query  A QueryInterface object. [optional]
```

Comment:
If its nullable then the docs should state that correctly as a type. 

```suggestion
	 * @param   mixed|null  $query  A QueryInterface object
```